sqlmodel_repository/base_repository.py

```diff
@@ -14,9 +16,24 @@ class BaseRepository(Generic[GenericEntity], ABC):
     """Abstract base class for all repositories"""
 
     entity: Type[GenericEntity]
+    logger: WriteLogger
+    sensitive_attribute_keys: list[str] = []
```

Comment:
those are class attributes, setting this in the `__init__` will cause unexpected behaviour because they override it for all instances of the class 😅 I would just remove them here and only set them in the `__init__`.

For the `entity` attribute it's kind of fine because it will always be the same